lib/index.d.ts

```diff
@@ -0,0 +1,43 @@
+import {Request} from 'express';
+import * as passport from 'passport';
+
+export interface StrategyOptions {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback?: true;
+  scope?: string[];
+}
+
+export interface StrategyOptionsWithRequest {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback: true;
```

Comment:
This is where things get slightly more complex. We define this very similar type because if we provide `true` for passReqToCallback, it means we must have a callback. So this gets combined with the `VerifyFunctionWithRequest` interface further down in our constructor definitions.